src/main/java/com/bakdata/kafka/util/ImprovedAdminClient.java

```diff
@@ -56,13 +57,14 @@ private ImprovedAdminClient(@NonNull final Properties properties,
             "%s must be specified in properties", AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG);
         this.properties = new Properties(properties);
         this.adminClient = AdminClient.create(properties);
-        this.schemaRegistryUrl = schemaRegistryUrl;
+        this.schemaRegistryClient =
+            schemaRegistryUrl == null ? null : createSchemaRegistryClient(this.properties, schemaRegistryUrl);
         this.timeout = timeout;
     }
 
     public Optional<SchemaClient> getSchemaClient() {
-        if (nonNull(this.schemaRegistryUrl)) {
-            return Optional.of(new SchemaClient(createSchemaRegistryClient(this.properties, this.schemaRegistryUrl)));
+        if (nonNull(this.schemaRegistryClient)) {
```

Comment:
`return Optional.ofNullable(this.schemaRegistryClient).map(SchemaClient::new)`

Reply:
Damn 🤯